[ADD Project Task Date Shift Module

patrickrwilson opened the pull request:
Adds the Project Task Date Shift Module. This module provide the optional ability to shift the dates on tasks when the project start date changes.

Original Discussion #500

patrickrwilson committed:
[ADD] Project Task Date Shift Module

Adds the Project Task Date Shift Module to give the option to shift task dates if the project date changes.

[FIX] PEP8 Errors

Revert "[FIX] PEP8 Errors"

This reverts commit 90f9cd939a72ece1fbfd8f42901cf479a66ac10c.

[FIX] PEP8 Formatting

[FIX] PEP8 Fixes

More PEP8 fixes that are causing the checks to fail.

[UPD] Suggested Changes

Updated with some suggested changes from other submitted modules.

emagdalenaC2i commented on project_task_date_shift/README.rst:
```diff
@@ -0,0 +1,82 @@
+
+This module is part of the OCA/project on GitHub.
+
+You are welcome to contribute. To learn how please visit https://odoo-community.org/page/Contribute.
```
You can update README follw the folder readme structure, see https://github.com/OCA/maintainer-tools/tree/master/template/module

emagdalenaC2i commented on project_task_date_shift/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+    'name': "Project Task Date Shift",
+
+    'summary': """
+        Project Task Date Shift""",
```
Delete extra "
I think it should be 'summary': "Project Task Date Shift",

emagdalenaC2i commented on project_task_date_shift/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+
+    'author': "Patrick Wilson, Odoo Community Association (OCA)",
+    'website': "https://github.com/OCA/project",
+
```
delete blank lines

emagdalenaC2i commented on project_task_date_shift/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+
+    # any module necessary for this one to work correctly
```
delete comments

emagdalenaC2i commented on project_task_date_shift/models/project.py:
```diff
@@ -0,0 +1,84 @@
+    _inherit = 'project.project'
+
+    old_start_date = fields.Date(string='Old Start Date',
+                                 help="Used by the Shift Dates function. \
```
I think this character is wrong \^M

patrickrwilson replied:
I don't know of another way to break the long help description up to be pep8 compliant and '\' seemed to work for all the other modules I've built. Any other suggestion?

emagdalenaC2i replied:
I think you don't need any extra character like in this example:
        help="For records searched by name (taxes, fiscal "
             "positions), the template name will be matched against the "
             "record name on this language.",

patrickrwilson replied:
Thanks, i've updated the code and it seems to be working correctly.

emagdalenaC2i commented on project_task_date_shift/views/project_task.xml:
```diff
@@ -0,0 +1,48 @@
+    </field>
+  </record>
+
+  <!-- Gantt View -->
```
Gantt View is a Timeline view?

patrickrwilson replied:
I changed it to <!-- Gantt/Timeline View --> to be more clear.

patrickrwilson committed:
[UPD] Review Changes

Updated Manifest and added Readme folder as well as some other suggested changes from reviewers.

patrickrwilson committed:
[UPD] Suggested Changes

Additional changes suggested by reviewers.

rafaelbn commented:
Please attend to an existing module https://github.com/OCA/project/tree/10.0/project_recalculate

cc @pedrobaeza

patrickrwilson commented:
@rafaelbn I'm not confident enough that my programming skills are advanced enough to convert that module to v12. If it's preferred by the community to convert that module since it's more advanced with looking at working hours, then feel free to cancel this pr.

rafaelbn commented:
@pedrobaeza when will we migrate https://github.com/OCA/project/tree/10.0/project_recalculate to v12?

pedrobaeza commented:
Not yet, but the module should be very similar

max3903 commented:
Shall we close this PR?

patrickrwilson commented:
I think so, seems it would be best to migrate the older module.
